engine.py

```diff
@@ -724,7 +725,7 @@ def _handle_active_document_change(self, active_document_path):
         # Make sure we have a properly-encoded string for the path. We can
         # possibly get a file path/name that contains unicode, and we don't
         # want to deal with that later on.
-        if isinstance(active_document_path, unicode):
+        if isinstance(active_document_path, six.text_type):
             active_document_path = active_document_path.encode("utf-8")
```

Comment:
This will fail in Python 3, as encode does not exist on `str`. You need to call six.ensure_str